Solutions/Azure Active Directory/Analytic Rules/AdminPromoAfterRoleMgmtAppPermissionGrant.yaml

```diff
@@ -24,54 +24,70 @@ relevantTechniques:
 tags:
   - SimuLand
 query: |
+  let query_frequency = 1h;
+  let query_period = 2h;
   AuditLogs
-  | where LoggedByService =~ "Core Directory"
-  | where Category =~ "ApplicationManagement"
-  | where AADOperationType =~ "Assign"
-  | where ActivityDisplayName =~ "Add app role assignment to service principal"
-  | mv-expand TargetResources
-  | mv-expand TargetResources.modifiedProperties
-  | extend displayName_ = tostring(TargetResources_modifiedProperties.displayName)
-  | where displayName_ =~ "AppRole.Value"
-  | extend AppRole = tostring(parse_json(tostring(TargetResources_modifiedProperties.newValue)))
-  | where AppRole has "RoleManagement.ReadWrite.Directory"
-  | extend InitiatingApp = tostring(parse_json(tostring(InitiatedBy.app)).displayName)
-  | extend Initiator = iif(isnotempty(InitiatingApp), InitiatingApp, tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName))
-  | extend Target = tostring(parse_json(tostring(TargetResources.modifiedProperties[4].newValue)))
-  | extend TargetId = tostring(parse_json(tostring(TargetResources.modifiedProperties[3].newValue)))
-  | project TimeGenerated, OperationName, Initiator, Target, TargetId, Result
-  | join kind=innerunique (
-    AuditLogs
-    | where LoggedByService =~ "Core Directory"
-    | where Category =~ "RoleManagement"
-    | where AADOperationType in ("Assign", "AssignEligibleRole")
-    | where ActivityDisplayName has_any ("Add eligible member to role", "Add member to role")
-    | mv-expand TargetResources
-    | mv-expand TargetResources.modifiedProperties
-    | extend displayName_ = tostring(TargetResources_modifiedProperties.displayName)
-    | where displayName_ =~ "Role.DisplayName"
-    | extend RoleName = tostring(parse_json(tostring(TargetResources_modifiedProperties.newValue)))
-    | where RoleName contains "Admin"
-    | extend Initiator = tostring(parse_json(tostring(InitiatedBy.app)).displayName)
-    | extend InitiatorId = tostring(parse_json(tostring(InitiatedBy.app)).servicePrincipalId)
-    | extend TargetUser = tostring(TargetResources.userPrincipalName)
-    | extend Target = iif(isnotempty(TargetUser), TargetUser, tostring(TargetResources.displayName))
-    | extend TargetType = tostring(TargetResources.type)
-    | extend TargetId = tostring(TargetResources.id)
-    | project TimeGenerated, OperationName,  RoleName, Initiator, InitiatorId, Target, TargetId, TargetType, Result
-  ) on $left.TargetId == $right.InitiatorId
-  | extend TimeRoleMgGrant = TimeGenerated, TimeAdminPromo = TimeGenerated1, ServicePrincipal = Initiator1, ServicePrincipalId = InitiatorId,
-    TargetObject = Target1, TargetObjectId = TargetId1, TargetObjectType = TargetType
-  | where TimeRoleMgGrant < TimeAdminPromo
-  | project TimeRoleMgGrant, TimeAdminPromo, RoleName, ServicePrincipal, ServicePrincipalId, TargetObject, TargetObjectId, TargetObjectType
+  | where TimeGenerated > ago(query_frequency)
```

Comment:
@devikamehra I believe other detections in AuditLogs do not use ingestion_time(), I don't think this detection has problems because of it, and if AuditLogs is ingested below 5min after TimeGenerated, then there shouldn't be any problems. Sentinel Analytics Scheduling accounts for these 5min between generation and ingestion.